Toolset/libraries/revidelibrary.8.livecodescript

```diff
@@ -1601,33 +1658,58 @@ on revIDEMessageSend pMessage, pEffectedTarget
       set the itemdel to ":"
       local tMessage
       put item 1 of pMessage into tMessage
-
-      switch tMessage
-         case "idePreferenceChanged"
-         case "ideToggleChanged"
-            dispatch tMessage to tObjectID with item 2 of pMessage
-            break
-         case "ideFindMoreWidgets"
-            dispatch pMessage to tObjectID with pEffectedTarget
-            break
-         case "ideNewStack"
-            try
-               send pMessage && pEffectedTarget to tObjectID in 10 milliseconds
-            catch tError
-            end try
-            break
-         case "idePropertyChanged"
-         case "ideMoveStack"
-            dispatch pMessage to tObjectID with pEffectedTarget
-            break
-         default
-            dispatch pMessage to tObjectID with pEffectedTarget
-            break
-      end switch
-
-      --put "SENT" && the milliseconds && pMessage && "sent to" && tObjectID & return & line 1 to 20 of msg into msg
+      
+      # Construct a dispatch command with all the desired parameters
+      local tDoString, tParams, tErrors
+      put "dispatch pMessage to tObjectID" into tDoString
+      
+      # If this is a parameterised message, add the parameter as the first arg
+      if the number of items of pMessage > 1 then
+         put " with item 2 of pMessage" into tParams
+      end if
+      
+      repeat with tParam = 1 to the number of elements in pParamsArray
+         if tParams is empty then
+            put " with pParamsArray[" & tParam & "]" into tParams
+         else
+            put ", pParamsArray[" & tParam & "]" after tParams
+         end if
+      end repeat
+      put tParams after tDoString
+      
+      # Ensure any error doesn't prevent subsequent subscribers from receiving messages
+      try
+         do tDoString
+      catch tError
+         if tErrors is empty then
+            put tError into tErrors
+         else
+            put return & tError after tErrors
+         end if
+      end try
```

Comment:
Yeah - If `the params` escaped quotes and return characters then it might be possible to do something with that (i.e. `dispatch <message> with word 3 to -1 of the params` or something like that)